app/assets/javascripts/discourse/app/initializers/welcome-topic-banner.js

```diff
@@ -0,0 +1,18 @@
+export default {
+  name: "welcome-topic-banner",
+  after: "message-bus",
+
+  initialize(container) {
+    const messageBus = container.lookup("service:message-bus");
+    if (!messageBus) {
+      return;
+    }
+
+    const site = container.lookup("service:site");
+    if (site.get("show_welcome_topic_banner")) {
```

Comment:
I think we can improve things further here. We do not even need to look up the messageBus service when `show_welcome_topic_banner` is false.

Reply:
When `show_welcome_topic_banner` is `false` then the messageBus service will not be subscribed per the above logic.

Reply:
Made the change.